tests/js/client/aql/aql-optimizer-push-limit-into-index.js

```diff
@@ -0,0 +1,187 @@
+/*jshint globalstrict:false, strict:false, maxlen: 500 */
+/*global assertTrue, assertEqual, assertNotEqual, assertNull, assertFalse */
+
+// //////////////////////////////////////////////////////////////////////////////
+// / DISCLAIMER
+// /
+// / Copyright 2014-2024 ArangoDB GmbH, Cologne, Germany
+// / Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+// /
+// / Licensed under the Business Source License 1.1 (the "License");
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     https://github.com/arangodb/arangodb/blob/devel/LICENSE
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is ArangoDB GmbH, Cologne, Germany
+// //////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const internal = require("internal");
+const errors = internal.errors;
+const db = require("@arangodb").db;
+const helper = require("@arangodb/aql-helper");
+const assertQueryError = helper.assertQueryError;
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite
+////////////////////////////////////////////////////////////////////////////////
+
+function optimizerPushLimitIntoIndexTestSuite () {
+  let c;
+  return {
+    setUpAll : function () {
+      db._drop("UnitTestsCollection");
+      c = db._create("UnitTestsCollection"); 
+      var docs = []; 
+      for (i = 0; i < 1000; ++i) { 
+        docs.push({ 
+          _key: "test" + i, 
+          date_created: "20240613" + i, 
+          license: (i <= 700 ? "cc-by-sa" : (i < 900 ? "cc-by-nc" : "foo")) 
+        }); 
+      }
+      c.insert(docs);
+      c.ensureIndex({ name: "license-date", type: "persistent", fields: ["license", "date_created"] }); 
+    },
+    tearDownAll : function () {
+      db._drop("UnitTestsCollection");
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief limit pushed into index node with applicable sorting conditions
+////////////////////////////////////////////////////////////////////////////////
+
+    testPushLimitIntoIndexRuleApplicableSortConditions : function () {
+      let queries = [
+        "FOR i IN " + c.name() + " FILTER i.license IN ['cc-by-sa', 'cc-by-nc', 'foo'] SORT i.date_created ASC LIMIT 100 RETURN i._key",
+        "FOR i IN " + c.name() + " FILTER i.license IN ['cc-by-sa', 'cc-by-nc', 'foo'] SORT i.date_created DESC LIMIT 100 RETURN i._key",
+      ];
+
+      for (var i = 0; i < queries.length; ++i) {
```

Comment:
for this and all following occurrences. Alternatively, you can use `queries.forEach(fn)`, which is even better.
```suggestion
      for (let i = 0; i < queries.length; ++i) {
```